app/src/main/java/pl/hypeapp/wykopolka/presenter/DemandQueuePresenter.java

```diff
@@ -72,6 +72,56 @@ public void onTransferBook(int position) {
         getView().showTransferDialog(bookPendingUserPair);
     }
 
+    public void onNotificationAction(Pair<Book, PendingUser> bookPendingUserPair) {
+        if (bookPendingUserPair.second.getWasCalled()) {
+            showMikroblogEntry(bookPendingUserPair.second.getCallId());
+        } else {
+            notifyUserOnMikroblog(bookPendingUserPair.second.getWishId());
+        }
+    }
+
+    private void showMikroblogEntry(String callId) {
+        getView().showMikroblogEntry(callId);
+    }
+
+    private void notifyUserOnMikroblog(String wishId) {
+        getView().setDialogInLoading();
+        String sign = HashUtil.generateApiSign(mAccountKey, wishId);
```

Comment:
Your presenter have one the major limitations - contains information about Rx library or other low-end library who returns information about external API or other data provider. In clean code you should hide specify implementation by abstraction, this practise call `Dependency Inversion` or `Inversion of Control`. 

Correct Presenter should contains interface who is a data bus between presenter layer and data layer. Specify implementation this interface should be in the data layer.